File: ArchitectureRefactoringHelper/Repository/Models/Recommendation/ApproachRecommendationRequest.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
using Repository.Models.Database;

namespace Repository.Models.Recommendation;

public class ApproachRecommendationRequest
{
    public ApproachRecommendationRequest()
    {
        DomainArtifactInformation = new List<AttributeRecommendationInformation<DomainArtifactInput>>();
        RuntimeArtifactInformation = new List<AttributeRecommendationInformation<RuntimeArtifactInput>>();
        ModelArtifactInformation = new List<AttributeRecommendationInformation<ModelArtifactInput>>();
        ExecutableInformation = new List<AttributeRecommendationInformation<ExecutableInput>>();
        QualityInformation = new List<AttributeRecommendationInformation<Quality>>();
        QualitySublevelInformation = new List<AttributeRecommendationInformation<QualitySublevel>>();
        DirectionInformation = new List<AttributeRecommendationInformation<Direction>>();
        AutomationLevelInformation = new List<AttributeRecommendationInformation<AutomationLevel>>();
        AnalysisTypeInformation = new List<AttributeRecommendationInformation<AnalysisType>>();
        TechniqueInformation = new List<AttributeRecommendationInformation<Technique>>();
        ArchitectureInformation = new List<AttributeRecommendationInformation<Architecture>>();
        ServiceTypeInformation = new List<AttributeRecommendationInformation<ServiceType>>();
        ResultsQualityInformation = new List<AttributeRecommendationInformation<ResultsQuality>>();
        ToolSupportInformation = new List<AttributeRecommendationInformation<ToolSupport>>();
        AccuracyPrecisionInformation = new List<AttributeRecommendationInformation<AccuracyPrecision>>();
        ValidationMethodInformation = new List<AttributeRecommendationInformation<ValidationMethod>>();
    }

    [Required]
    [JsonPropertyName("domainArtifactInformation")]
    public List<AttributeRecommendationInformation<DomainArtifactInput>> DomainArtifactInformation { get; set; }

    [Required]
    [JsonPropertyName("runtimeArtifactInformation")]
    public List<AttributeRecommendationInformation<RuntimeArtifactInput>> RuntimeArtifactInformation { get; set; }

    [Required]
    [JsonPropertyName("modelArtifactInformation")]
    public List<AttributeRecommendationInformation<ModelArtifactInput>> ModelArtifactInformation { get; set; }

    [Required]
    [JsonPropertyName("executableInformation")]
    public List<AttributeRecommendationInformation<ExecutableInput>> ExecutableInformation { get; set; }

    [Required]
    [JsonPropertyName("qualityInformation")]
    public List<AttributeRecommendationInformation<Quality>> QualityInformation { get; set; }

    [Required]
    [JsonPropertyName("qualitySublevelInformation")]
    public List<AttributeRecommendationInformation<QualitySublevel>> QualitySublevelInformation { get; set; }

    [Required]
    [JsonPropertyName("directionInformation")]
    public List<AttributeRecommendationInformation<Direction>> DirectionInformation { get; set; }

    [Required]
    [JsonPropertyName("automationLevelInformation")]
    public List<AttributeRecommendationInformation<AutomationLevel>> AutomationLevelInformation { get; set; }

    [Required]
    [JsonPropertyName("analysisTypeInformation")]
    public List<AttributeRecommendationInformation<AnalysisType>> AnalysisTypeInformation { get; set; }

    [Required]
    [JsonPropertyName("techniqueInformation")]
    public List<AttributeRecommendationInformation<Technique>> TechniqueInformation { get; set; }

    [Required]
    [JsonPropertyName("architectureInformation")]
    public List<AttributeRecommendationInformation<Architecture>> ArchitectureInformation { get; set; }

    [Required]
    [JsonPropertyName("serviceTypeInformation")]
    public List<AttributeRecommendationInformation<ServiceType>> ServiceTypeInformation { get; set; }

    [Required]
    [JsonPropertyName("resultsQualityInformation")]
    public List<AttributeRecommendationInformation<ResultsQuality>> ResultsQualityInformation { get; set; }

    [Required]
    [JsonPropertyName("toolSupportInformation")]
    public List<AttributeRecommendationInformation<ToolSupport>> ToolSupportInformation { get; set; }

    [Required]
    [JsonPropertyName("accuracyPrecisionInformation")]
    public List<AttributeRecommendationInformation<AccuracyPrecision>> AccuracyPrecisionInformation { get; set; }

    [Required]
    [JsonPropertyName("validationMethodInformation")]
    public List<AttributeRecommendationInformation<ValidationMethod>> ValidationMethodInformation { get; set; }
}

File: UI/api/repository/models/approach-recommendation-request.ts
/* tslint:disable */
/* eslint-disable */
import { AccuracyPrecisionAttributeRecommendationInformation } from './accuracy-precision-attribute-recommendation-information';
import { AnalysisTypeAttributeRecommendationInformation } from './analysis-type-attribute-recommendation-information';
import { ArchitectureAttributeRecommendationInformation } from './architecture-attribute-recommendation-information';
import { AutomationLevelAttributeRecommendationInformation } from './automation-level-attribute-recommendation-information';
import { DirectionAttributeRecommendationInformation } from './direction-attribute-recommendation-information';
import { DomainArtifactInputAttributeRecommendationInformation } from './domain-artifact-input-attribute-recommendation-information';
import { ExecutableInputAttributeRecommendationInformation } from './executable-input-attribute-recommendation-information';
import { ModelArtifactInputAttributeRecommendationInformation } from './model-artifact-input-attribute-recommendation-information';
import { QualityAttributeRecommendationInformation } from './quality-attribute-recommendation-information';
import { QualitySublevelAttributeRecommendationInformation } from './quality-sublevel-attribute-recommendation-information';
import { ResultsQualityAttributeRecommendationInformation } from './results-quality-attribute-recommendation-information';
import { RuntimeArtifactInputAttributeRecommendationInformation } from './runtime-artifact-input-attribute-recommendation-information';
import { ServiceTypeAttributeRecommendationInformation } from './service-type-attribute-recommendation-information';
import { TechniqueAttributeRecommendationInformation } from './technique-attribute-recommendation-information';
import { ToolSupportAttributeRecommendationInformation } from './tool-support-attribute-recommendation-information';
import { ValidationMethodAttributeRecommendationInformation } from './validation-method-attribute-recommendation-information';
export interface ApproachRecommendationRequest {
  accuracyPrecisionInformation: Array<AccuracyPrecisionAttributeRecommendationInformation>;
  analysisTypeInformation: Array<AnalysisTypeAttributeRecommendationInformation>;
  architectureInformation: Array<ArchitectureAttributeRecommendationInformation>;
  automationLevelInformation: Array<AutomationLevelAttributeRecommendationInformation>;
  directionInformation: Array<DirectionAttributeRecommendationInformation>;
  domainArtifactInformation: Array<DomainArtifactInputAttributeRecommendationInformation>;
  executableInformation: Array<ExecutableInputAttributeRecommendationInformation>;
  modelArtifactInformation: Array<ModelArtifactInputAttributeRecommendationInformation>;
  qualityInformation: Array<QualityAttributeRecommendationInformation>;
  qualitySublevelInformation: Array<QualitySublevelAttributeRecommendationInformation>;
  resultsQualityInformation: Array<ResultsQualityAttributeRecommendationInformation>;
  runtimeArtifactInformation: Array<RuntimeArtifactInputAttributeRecommendationInformation>;
  serviceTypeInformation: Array<ServiceTypeAttributeRecommendationInformation>;
  techniqueInformation: Array<TechniqueAttributeRecommendationInformation>;
  toolSupportInformation: Array<ToolSupportAttributeRecommendationInformation>;
  validationMethodInformation: Array<ValidationMethodAttributeRecommendationInformation>;
}
